platform-server-kernel/src/main/java/uno/cod/platform/server/core/filter/TokenHelper.java

```diff
@@ -0,0 +1,42 @@
+package uno.cod.platform.server.core.filter;
+
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.crypto.codec.Base64;
+
+import java.io.IOException;
+
+/**
+ * Created by drogetzer on 13.07.2016.
+ */
+public class TokenHelper {
+
+    /**
+     * Decodes the header into a username and password.
+     *
+     * @throws BadCredentialsException if the Basic header is not present or is not valid
```

Comment:
That's wrong. The method will `throw BadCredentialsexception(...);` iff:

 * `Base64.decode` throws an `IlegalArgumentException`
 * The decoded `String` does not contain the character `:`.

So the case "is not valid Bas64" is fine, but "Basic header is not present" is not. What even is the "Basic header"? If the header is empty or `null`, will it be valid Base64?

The throwing case of the decoded header not containing `:` is not mentioned in any way.

I know that this has been introduced by @Exordian [here](https://github.com/coduno/platform/commit/00ec02dd8d47ae85c49bae48df39d479b209dcfd#diff-08b8bac60e80cc76c7c139f7abe0e7d3R84) but nevertheless we could improve it.

I'd suggest the following:

```java
/**
 * Decodes given {@link String} using Base64 and splits it by the colon character.
 *
 * @throws BadCredentialsException if <code>header</code> is <code>null</null> or it's decoded form does not contain a colon.
 * @return the two parts of the decoded string
 * @see <a href="https://tools.ietf.org/html/rfc7235#section-2">RFC 7235, Section 2</a>
 */
```

Furthermore, I'd suggest throwing an `IllegalArgumentException` instead of a `BadCredentialsException` as this helper method is not obviously part of an authentication filter.